demos/jans-tarp/src/options/oidcClientDetails.tsx

```diff
@@ -0,0 +1,264 @@
+import React, { useState, useEffect } from 'react'
+import { v4 as uuidv4 } from 'uuid';
+import axios from 'axios';
+import qs from 'qs';
+import './options.css'
+import { WindmillSpinner } from 'react-spinner-overlay'
+
+const OIDCClientDetails = (data) => {
+    const [additionalParam, setAdditionalParam] = useState("");
+    const [loading, setLoading] = useState(false);
+
+    function customLaunchWebAuthFlow(options, callback) {
+        var requestId = Math.random().toString(36).substring(2);
+        var authUrl = options.url + "&state=" + requestId;
+        chrome.tabs.create({ url: authUrl }, function (tab) {
+            var intervalId = setInterval(function () {
+                chrome.tabs.get(tab.id, function (currentTab) {
+                    if (!currentTab) {
+                        clearInterval(intervalId);
+                        chrome.tabs.remove(tab.id);
+                        callback(undefined, new Error('Authorization tab was closed.'));
+                    }
+                });
+            }, 1000);
+
+            chrome.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
+                if (!!chrome.runtime.lastError) {
+                    chrome.tabs.remove(tabId);
+                    callback(undefined, chrome.runtime.lastError);
+                }
+                if (tabId === tab?.id && changeInfo?.status === "complete") {
+                    chrome.tabs.sendMessage(tab.id, { requestId: requestId }, function (response) {
+                        clearInterval(intervalId);
+                        chrome.tabs.query({ active: true, lastFocusedWindow: true }, tabs => {
+                            let url = tabs[0].url;
+                            const urlParams = new URLSearchParams(new URL(url).search)
+                            const code = urlParams.get('code')
+                            if (code != null) {
+                                callback(url, undefined);
+                                chrome.tabs.remove(tab.id);
+                            }
+                        });
+                    });
+                }
+                //chrome.tabs.onUpdated.removeListener(tabUpdatedListener);
+            })
+        });
+    }
+
+    async function triggerCodeFlowButton() {
+        setLoading(true);
+        const redirectUrl = chrome.identity.getRedirectURL()
+        const { secret, hashed } = await generateRandomChallengePair();
+        chrome.storage.local.get(["oidcClient"]).then(async (result) => {
+            if (result.oidcClient != undefined) {
```

Comment:
please add optional chaining before accessing `oidcClient` value and we can change the condition to just `!result.oidcClient` which will handle where result may not exist instead of explicit value `undefined`